test/unit/test_keyring_raw_aes.py

```diff
@@ -0,0 +1,343 @@
+
+    generated_flag_count = 0
+    encrypted_flag_count = 0
+
+    for keyring_trace in test.keyring_trace:
+        if (
+            keyring_trace.wrapping_key.key_info == _KEY_ID
+            and KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY in keyring_trace.flags
+        ):
+            # Check keyring trace contains KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
+            generated_flag_count += 1
+        if KeyringTraceFlag.WRAPPING_KEY_ENCRYPTED_DATA_KEY in keyring_trace.flags:
+            encrypted_flag_count += 1
+
+    assert generated_flag_count == 1
+
+    assert len(test.encrypted_data_keys) == original_number_of_encrypted_data_keys + 1
+
+    assert encrypted_flag_count == 1
+
+
+def test_on_decrypt_when_data_key_given(patch_decrypt_data_key):
+    test_raw_aes_keyring = RawAESKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+    test_raw_aes_keyring.on_decrypt(
+        decryption_materials=get_decryption_materials_with_data_encryption_key(),
+        encrypted_data_keys=[
+            EncryptedDataKey(
+                key_provider=MasterKeyInfo(provider_id=_PROVIDER_ID, key_info=_KEY_ID),
+                encrypted_data_key=b"\xde^\x97\x7f\x84\xe9\x9e\x98\xd0\xe2\xf8\
+                                   xd5\xcb\xe9\x7f.}\x87\x16,\x11n#\xc8p"
+                b"\xdb\xbf\x94\x86*Q\x06\xd2\xf5\xdah\x08\xa4p\x81\xf7\xf4G\
+                                   x07FzE\xde",
+            )
+        ],
+    )
+    assert not patch_decrypt_data_key.called
+
+
+def test_keyring_trace_on_decrypt_when_data_key_given():
+    test_raw_aes_keyring = RawAESKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+    test = test_raw_aes_keyring.on_decrypt(
+        decryption_materials=get_decryption_materials_with_data_encryption_key(),
+        encrypted_data_keys=[
+            EncryptedDataKey(
+                key_provider=MasterKeyInfo(provider_id=_PROVIDER_ID, key_info=_KEY_ID),
+                encrypted_data_key=b"\xde^\x97\x7f\x84\xe9\x9e\x98\xd0\xe2\xf8\
+                                   xd5\xcb\xe9\x7f.}\x87\x16,\x11n#\xc8p"
+                b"\xdb\xbf\x94\x86*Q\x06\xd2\xf5\xdah\x08\xa4p\x81\xf7\xf4G\
+                                   x07FzE\xde",
+            )
+        ],
+    )
+    for keyring_trace in test.keyring_trace:
+        if keyring_trace.wrapping_key.key_info == _KEY_ID:
+            # Check keyring trace does not contain KeyringTraceFlag.WRAPPING_KEY_DECRYPTED_DATA_KEY
+            assert KeyringTraceFlag.WRAPPING_KEY_DECRYPTED_DATA_KEY not in keyring_trace.flags
+
+
+def test_on_decrypt_when_data_key_and_edk_not_provided(patch_decrypt_data_key):
```

Comment:
Let's combine this and `test_on_decrypt_when_data_key_not_provided_and_edk_not_in_keyring` into a single test with two parameterized values for `encrypted_data_keys`.